frontend_tests/zjsunit/namespace.js

```diff
@@ -96,15 +96,31 @@ exports.with_overrides = function (test_function) {
 
     const restore_callbacks = [];
     const unused_funcs = new Map();
+    const funcs = new Map();
 
     const override = function (module, func_name, f) {
         if (typeof f !== "function") {
             throw new TypeError("You can only override with a function.");
         }
 
+        if (!funcs.has(module)) {
+            funcs.set(module, new Map());
+        }
+
+        if (funcs.get(module).has(func_name)) {
+            // Prevent overriding the same function twice, so that
+            // it's super easy to reason about our logic to restore
+            // the original function.  Usually if somebody sees this
+            // error, it's a symptom of not breaking up tests enough.
+            throw new Error("You can only override a function one time.");
```

Comment:
The logic to restore the original function was fine, because the `restore_callbacks` are executed in reverse order.

Reply:
Yep, i wrote it that way, but it felt brittle.  The bigger point, which I probably could have phrased more carefully in the comments, is that it’s almost always a code smell if you’re doing it multiple times in a test.

Reply:
0d68a23066245c07dc58ffda8f48483e13e98ae0